Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -477,7 +477,8 @@ bool Connection::platformPrepareForOpen()
 void Connection::invalidate()
 {
     m_isValid = false;
-    if (!m_client)
+    auto client = m_client.get();
```

Comment:
Would these changes be possible to make with just holding the m_client as weakptr?